firefox-ios/Client/Telemetry/GleanWrapper.swift

```diff
@@ -7,21 +7,103 @@ import Glean
 
 protocol GleanWrapper {
     func handleDeeplinkUrl(url: URL)
-    func submitPing()
     func setUpload(isEnabled: Bool)
+    func submitPing()
+
+    // MARK: Glean Metrics
+
+    func submitEventMetricType<ExtraObject>(event: EventMetricType<ExtraObject>,
+                                            extras: EventExtras) where ExtraObject: EventExtras
+    func submitEventMetricType<NoExtras>(event: EventMetricType<NoExtras>) where NoExtras: EventExtras
+    func submitCounterMetricType(event: CounterMetricType)
+    func submitStringMetricType(event: StringMetricType, value: String)
+    func submitLabeledMetricType(event: LabeledMetricType<CounterMetricType>, value: String)
+    func submitBooleanMetricType(event: BooleanMetricType, value: Bool)
+    func submitQuantityMetricType(event: QuantityMetricType, value: Int64)
+
+    func addToNumeratorRateMetricType(event: RateMetricType, amount: Int32)
+    func addToDenominatorRateMetricType(event: RateMetricType, amount: Int32)
```

Comment:
nit: instead of using `event`, what about using `metricType` for the param names? To avoid confusion that there are event types and other types in telemetry. 

Also, what about modifying the naming to align more on the method calls such as `.add`, `.record`, `.set`.

Here's an example:
```
    func recordEvent<ExtraObject>(for metricType: EventMetricType<ExtraObject>,
                                            extras: EventExtras) where ExtraObject: EventExtras
    func addCounter(for metricType: CounterMetricType)
    func setBoolean(for metricType: BooleanMetricType, value: Bool)
```